### PR TITLE
Allow injecting raw configurations

### PR DESCRIPTION
**Pull request by fedepaol**
To leverage the frr instance and explore new use cases, we allow injecting raw configs that are appended after the rendered one.
Given that multiple configurations might be applied to the same node, a priority field drives which config should go first.

### CODE REVIEW

File: api/v1beta1/frrconfiguration_types.go [hunk 1]
```diff
@@ -26,19 +26,29 @@ type FRRConfigurationSpec struct {
...
 	// whose labels match the specified selectors.
 	// When it is not specified all nodes will attempt to apply this config.
 	// +optional
 	NodeSelector metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
+
+type RawConfig struct {
+	// Sets the order with this configuration is appended to the
+	// bottom of the rendered configuration
+	Priority int `json:"priority,omitempty"`
```
Comment (oribon) [inline api/v1beta1/frrconfiguration_types.go:42]: does this have a value for the user (other than us using this for "deterministic order"?)

File: internal/controller/api_to_config.go [hunk 2]
```diff
@@ -346,3 +352,15 @@ func sortMapPtr[T any](toSort map[string]*T) []*T {
 	}
 	return res
 }
+
+func joinRawConfigs(raw []v1beta1.RawConfig) string {
+	sort.Slice(raw, func(i, j int) bool {
+		return raw[i].Priority > raw[j].Priority
```
Comment (oribon) [inline internal/controller/api_to_config.go:358]: this is nondeterministic if `raw` contains multiple with the same prio in a different order everytime (maybe it makes sense to sort the fromK8s cfgs first by name)?

File: internal/controller/api_to_config_test.go [hunk 3]
```diff
@@ -1309,6 +1309,81 @@ func TestConversion(t *testing.T) {
...
+						IPV4Prefixes: []string{},
+						IPV6Prefixes: []string{},
+					},
+				},
+				ExtraConfig: "foo\n",
+			},
+			err: nil,
+		},
+		{
+			name: "Single Router and double injection",
```
Comment (oribon) [inline internal/controller/api_to_config_test.go:1347]: is there any value in testing a multi line raw config?

File: e2etests/tests/raw_config.go [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+			nodes, err := k8s.Nodes(cs)
+			framework.ExpectNoError(err)
+
+			for _, c := range frrs {
+				ValidateFRRPeeredWithNodes(nodes, c, ipfamily.IPv4)
+			}
+			config.Spec.Raw.Config = nil
+			err = updater.Update(config)
+			framework.ExpectNoError(err)
```
Comment (oribon) [inline e2etests/tests/raw_config.go:129]: what does this do?

File: api/v1beta1/frrconfiguration_types.go (hunk 1 above)
Comment (fedepaol) [reply api/v1beta1/frrconfiguration_types.go:42]: well the order (and having the control of what config lands) is not only for us.
They might want to define the rotuer in one config and the rules in another, or what not.
It makes sense to control the order.

File: internal/controller/api_to_config_test.go (hunk 3 above)
Comment (fedepaol) [reply internal/controller/api_to_config_test.go:1347]: I can add

File: e2etests/tests/raw_config.go (hunk 4 above)
Comment (fedepaol) [reply e2etests/tests/raw_config.go:129]: Leftover, let me remove it

**Review by oribon — COMMENTED** [review]
small nit, other than that lgtm

File: api/v1beta1/frrconfiguration_types.go [hunk 5]
```diff
@@ -26,19 +26,29 @@ type FRRConfigurationSpec struct {
...
 	// When specified, the configuration will be considered only on nodes
 	// whose labels match the specified selectors.
 	// When it is not specified all nodes will attempt to apply this config.
 	// +optional
 	NodeSelector metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
+
+type RawConfig struct {
+	// Sets the order with this configuration is appended to the
+	// bottom of the rendered configuration
```
[the remark this answers is not in the document]
Comment (fedepaol) [reply api/v1beta1/frrconfiguration_types.go:41]: *before

[the remark this answers is not in the document]
Comment (fedepaol) [reply api/v1beta1/frrconfiguration_types.go:41]: done

**Review by oribon — APPROVED** [review]
lgtm

**Pull request merged**